src/uct/tcp/tcp_ep.c

```diff
@@ -370,24 +374,21 @@ uct_tcp_ep_create_socket_and_connect(uct_tcp_iface_t *iface,
         ep->fd = fd;
     }
 
-    status = uct_tcp_cm_conn_start(ep);
-    if (status != UCS_OK) {
-        goto err_ep_destroy;
+    ep->conn_retries++;
+    if (ep->conn_retries > iface->config.max_conn_retries) {
+        ucs_error("tcp_ep %p: reached maximum number of connection retries "
+                  "(%u)", ep, iface->config.max_conn_retries);
+        uct_tcp_ep_set_failed(ep);
```

Comment:
maybe unite uct_tcp_ep_set_failed here with the one inside uct_tcp_cm_conn_start